pkg/apis/config/feature_flags.go

```diff
@@ -203,13 +210,14 @@ type FeatureFlags struct {
 	Coschedule                               string `json:"coschedule,omitempty"`
 	EnableCELInWhenExpression                bool   `json:"enableCELInWhenExpression,omitempty"`
 	// EnableStepActions is a no-op flag since StepActions are stable
-	EnableStepActions            bool   `json:"enableStepActions,omitempty"`
-	EnableParamEnum              bool   `json:"enableParamEnum,omitempty"`
-	EnableArtifacts              bool   `json:"enableArtifacts,omitempty"`
-	DisableInlineSpec            string `json:"disableInlineSpec,omitempty"`
-	EnableConciseResolverSyntax  bool   `json:"enableConciseResolverSyntax,omitempty"`
-	EnableKubernetesSidecar      bool   `json:"enableKubernetesSidecar,omitempty"`
-	EnableWaitExponentialBackoff bool   `json:"enableWaitExponentialBackoff,omitempty"`
+	EnableStepActions                bool   `json:"enableStepActions,omitempty"`
+	EnableParamEnum                  bool   `json:"enableParamEnum,omitempty"`
+	EnableArtifacts                  bool   `json:"enableArtifacts,omitempty"`
+	DisableInlineSpec                string `json:"disableInlineSpec,omitempty"`
+	EnableConciseResolverSyntax      bool   `json:"enableConciseResolverSyntax,omitempty"`
+	EnableKubernetesSidecar          bool   `json:"enableKubernetesSidecar,omitempty"`
+	EnableWaitExponentialBackoff     bool   `json:"enableWaitExponentialBackoff,omitempty"`
+	DeprecatedEnableTektonOCIBundles *bool  `json:"enableTektonOCIBundles,omitempty"       yaml:"EnableTektonOCIBundles,omitempty"`
```

Comment:
```suggestion
    // DeprecatedEnableTektonOCIBundles is maintained for backward compatibility
    // to allow deletion of PipelineRuns created before v0.62.x.
    // This field is not used and can be removed in a future release
    // once we're confident old PipelineRuns have been cleaned up.
    // See issue #8359 for context.
	DeprecatedEnableTektonOCIBundles *bool  `json:"enableTektonOCIBundles,omitempty"       yaml:"enableTektonOCIBundles,omitempty"`
```

add a explanation comment and fix casing in the yaml tag